server/src/controllers/campaign.js

```diff
@@ -0,0 +1,97 @@
+const db = require("../../database/models/index");
+const BadRequestError = require("../errors/bad-request-error");
+const Op = db.Sequelize.Op;
+
+class CampaignController {
+  static getCampaigns = async (req, res) => {
+    const user = await db.User.findOne({
+      where: {
+        id: req.params.userId
+      }
+    });
+
+    if (!user) {
+      throw new BadRequestError("User does not exist");
+    }
+
+    const campaigns = await user.getCampaigns();
+
+    res.send(campaigns);
+  };
+
+  static getCampaign = async (req, res) => {
+    const campaign = await db.Campaign.findOne({
+      where: {
+        id: req.params.id
+      },
+      include: db.Prospect
+    });
+
+    if (!campaign) {
+      throw new BadRequestError("Campaign does not exist");
+    }
+
+    res.send(campaign);
+  };
+
+  static createCampaign = async (req, res) => {
+    const user = await db.User.findOne({
+      where: {
+        id: req.params.userId
+      }
+    });
+
+    if (!user) {
+      throw new BadRequestError("User does not exist");
+    }
+
+    const newCampaign = await db.Campaign.create({
+      name: req.body.name,
+      userId: user.id
+    });
+
+    res.status(201).send(newCampaign);
+  };
+
+  static addProspects = async (req, res) => {
+    const campaign = await db.Campaign.findOne({
+      where: {
+        id: req.params.id
+      }
+    });
+
+    if (!campaign) {
+      throw new BadRequestError("Campaign does not exist");
+    }
+
+    const prospects = await db.Prospect.findAll({
```

Comment:
Need to make sure these prospects are all owned by this user, maybe you can add that to query (get all prospects by id and user_id

Reply:
Updated to add only prospects that is owned by the user